HUD/hud_scraper.py

```diff
@@ -0,0 +1,177 @@
+import os
+import sys
+import json
+import pandas as pd
+from pymongo import MongoClient, TEXT, errors
+import random
+from tqdm import tqdm
+import re
+from collections import OrderedDict
+import requests
+from datetime import datetime, date
+
+if __package__:  # if script is being run as a module
+    from ..shelterapputils.utils import check_similarity, refresh_ngrams
+else:  # if script is being run as a file
+    _i = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    if _i not in sys.path:
+        # add parent directory to sys.path so utils module is accessible
+        sys.path.insert(0, _i)
+    del _i  # clean up global name space
+    from shelterapputils.utils import (
+        check_similarity, refresh_ngrams,
+        make_ngrams, locate_potential_duplicate,
+        distance, insert_services
+    )
+# Establish global variables
+client = MongoClient(
+    "mongodb+srv://" + os.environ.get('DBUSERNAME') + ":" + os.environ.get('PW')
+    + "@shelter-rm3lc.azure.mongodb.net/shelter?retryWrites=true&w=majority"
+)['shelter']
+
+
+def grab_data():
+    """Access hosted HUD csv file,
+       clean things up a bit and add code descriptions.
+
+    Returns:
+        DataFrame: Pandas DataFrame containing processed data
+    """
+    data_url = 'https://www.huduser.gov/portal/sites/default/files/xls/2019' \
+        '-Housing-Inventory-County-RawFile.xlsx'
+    df = pd.read_excel(
+        data_url,
+        usecols=[
+            'Row #', 'CocState', 'Geo Code', 'CoC', 'Coc\ID', 'HudNum', 'Organization ID',
+            'Organization Name', 'HMIS Org ID', 'Project ID', 'Target Population',
+            'Project Name', 'HMIS Project ID', 'HIC Date', 'Project Type',
+            'Bed Type', 'Victim Service Provider', 'address1',
+            'address2', 'city', 'state', 'zip', 'Total Beds', 'Updated On'
+        ]
+    )
+    df = df.drop_duplicates(
+        subset=[
+            'Organization Name', 'Project Name', 'address1', 'city', 'state', 'zip'
+        ],
+        ignore_index=True
+    ).reset_index(drop=True)
+    df['source'] = ['HUD'] * len(df)
+    df['service_type'] = ['SHELTER'] * len(df)
+    service_summaries = []
+    for i in tqdm(range(len(df))):
+        if df.loc[i, 'Project Type'] == 'ES':
+            if df.loc[i, 'Target Population'] is not None:
+                service_summaries.append('Domestic Violence Shelter')
+            else:
+                service_summaries.append('Emergency Shelter')
+        else:
+            service_summaries.append('Transitional Housing')
+    df['service_summary'] = service_summaries
+    return df
+
+
+def scrape_updated_date():
+    url = 'https://www.hudexchange.info/resource/3031/pit-and-hic-data-since-2007/'
```

Comment:
probably should have it in config file?

Reply:
I think it's fine to have a couple URLs in the script rather than making a whole config file